core/src/services/config.js

```diff
@@ -1,85 +1,96 @@
-import { isFunction, isPromise } from '../utilities/helpers.js';
+import { getConfigValueFromObjectAsync } from '../utilities/async-helpers';
+import { getConfigValueFromObject } from '../utilities/helpers';
 
-(function() {
-  window.Luigi = {
-    setConfig: configInput => {
-      window.Luigi.config = configInput;
-    },
-    getConfig: () => {
-      return window.Luigi.config;
-    }
-  };
-})();
-/*
- * Gets value of the given property on the given object.
- */
-export const getConfigValueFromObject = (object, property) => {
-  let propIndex = 0;
-  let nextValue = object;
-  const propertyPath = property.split('.');
-  while (nextValue && propIndex < propertyPath.length) {
-    nextValue = nextValue[propertyPath[propIndex++]];
+class LuigiConfigManager {
+  constructor() {
+    this.configReadyTimeout = {
+      valueMs: 2000,
+      id: undefined
+    };
+    this.config;
```

Comment:
What doest this line do? `this.config` stays undefined after `constructor()` finished so there's no need to "declare" this variable... Correct me if I'm wrong.

Reply:
removed